tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/RunMojo.java

```diff
@@ -145,12 +185,30 @@ private void deploy(BundleContext bundleContext) throws MojoExecutionException {
                 } else {
                     throw new MojoExecutionException("Packaging " + project.getPackaging() + " is not supported");
                 }
+            } else if (attachedFeatureFileExists) {
+                addFeaturesAttachmentAsFeatureRepository(featureService, attachedFeatureFile);
             } else {
                 throw new MojoExecutionException("Project artifact doesn't exist");
             }
         }
     }
 
+    void addFeatures(Object featureService) throws MojoExecutionException {
+    	if (featuresToInstall != null) {
+            try {
+            	Class<? extends Object> serviceClass = featureService.getClass();
+                Method installFeatureMethod = serviceClass.getMethod("installFeature", String.class);
+                String[] features = featuresToInstall.split(" *, *");
+                for (String feature : features) {
+                    installFeatureMethod.invoke(featureService, feature);
+                    Thread.sleep(1000L);
```

Comment:
Fair enough, let's keep like this as a first step. We will improve later. Thanks.